sklearn/utils/estimator_checks.py

```diff
@@ -314,7 +336,7 @@ def set_testing_parameters(estimator):
         # of components of the random matrix projection will be probably
         # greater than the number of features.
         # So we impose a smaller number (avoid "auto" mode)
-        estimator.set_params(n_components=1)
+        estimator.set_params(n_components=2)
```

Comment:
Yeah. Something like that. If that's feasible. Either that or something more explicit, like requiring a flag to be passed in.